### DISCUSSION TITLE
Is there a way to find out when the payment form has loaded/rendered?

### DISCUSSION THREAD

**Discussion by thecannabisapp** [weareseeed/react-square-web-payments-sdk#36] [Help] [answered]
I'm noticing that when I load the payment form, there's a delay between the screen loading vs form loading. Is there a way where I can find out when the <SquarePaymentsForm>  component has been rendered so I can load a spinner before to indicate the form is not ready?

**Comment by danestves** [comment]
Currently, we don't have anything like that to know the lifecycle of the wrapper, but it sounds really great to have something like `onReady` on every component, so you can do anything you want with that

Will speak with the team to see how we can do this and allow users to use it, when we have something to test I will pin you 😊

**Comment by danestves** [comment, marked as answer]
Hi @chinesehemp with the v3 we have implemented some loaders in some components of the wrapper, can you give it a try? https://github.com/weareseeed/react-square-web-payments-sdk/releases/tag/v3.0.0
